### ISSUE TITLE
Tab Sync results in "Failed to map parentid: tabs"

### ISSUE THREAD

**Issue by raslop** [floccusaddon/floccus#1234]
### Which version of floccus are you using?

4.17.0

### Sync method

WebDAV

### Which browser are you using? In case you are using the phone App, specify the Android or iOS version and device please.

Firefox (64 bit) 103.0.1 and 104.0.1 on Arch Linux
Kiwi Browser on 105.0.5195.33 on Android 11.

### What kind of WebDAV server are you using? (if relevant)

not relevant

### Describe the Bug

Since the auto-update to version 4.17.0 tab synchronisation always fails on every device with "Failed to map parentId: tabs". With previous version tab synchronisation worked fine for me.
I also tried to remove the tab sync account and re-add it, but the error occurred immediately in the newly created account.

(debug log uploaded to below mentioned URL (floccus-4.17.0-2022-08-11.log))

### Expected Behavior

Tab synchronisation succeeds.

### To Reproduce

Create new account with the following options:
1. webdav
2. url and credentials as needed, no passphrase
3. xbel format, synchronize tabs
4. upload only, embed local folder (auto-sync on, interval 15min)
Wait until first sync or initiate first sync manually and see the error.

### Debug log provided

- [X] I have provided a debug log file

**Comment by marcelklehr** [comment]
Mmh, I cannot replicate this. :/ Does this still happen?

**Comment by raslop** [comment]
Yes, it still occurs with my standard firefox profile.
Btw. this also happens with Kiwi Browser 105.0.5195.33 on Android 11.

However, I tried with a freshly created firefox (now 104.0.1) and was also not able to reproduce the error.

I tried to look into local storage of the floccus extension in firefox (via debug extensions), and tried to compare floccus tab-sync account data between the two firefox profiles. But wasn't able to spot anything suspicious (which is not surprising, as I have only little knowledge with web/app-developing).

Is there anything specific I could look for or investigate with my standard firefox profile?
I'm not too enthusiastic about migrating to a new firefox profile on every sync device.

**Comment by marcelklehr** [comment]
You could try uninstalling floccus and reinstalling, which should wipe the extension storage.

**Comment by raslop** [comment]
I thought of that too, but didn't try it so far, now I have and it didn't work.
Then I tried to to import my accounts to a newly created firefox profile, which resulted in the same error.
Then I moved away the corresponding xbel-file from the webdav server, and voila, the error is gone and uploading works again.
Btw. manually creating the tab-upload-only account in a fresh firefox profile with the same webdav-file results in the same error. Earlier today (see above) I tried with a different webdav file and couldn't reproduce the error. So the error is obviously related to the contents of the xbel-file on the webdav server.

While further investigating this, I noticed the following things:
1. Floccus does a HTTP GET for the xbel-file even for upload-only accounts, and somehow interprets the result. As the account is upload-only, I find the HTTP GET surprising. The file is overwritten anyway shortly after.
2. The difference between the existing file, uploaded from floccus 4.16.0, and the new file, from floccus 4.17.0, is that the 4.16.0 file has an empty "title" tag in the top-level "folder" (i.e. `<title/>`), while the 4.17.0 file has a non-empty title tag in the top-level folder, i.e. `<title>Window 0></title>`. Obviously floccus 4.17.0 doesn't like the empty title tag from earlier versions and bails out the error "Failed to map parentid: tabs".
3. On some other device I have this same webdav file as download-only account. This works fine, even with the empty "title" tag. So there seem to be different code-paths in floccus to interpret the xbel-files.
